src/events/broker.py

```diff
@@ -201,6 +201,9 @@ def _on_config_changed(self, event: EventBase) -> None:
         if self.model.relations.get(REL_NAME, None) and self.charm.unit.is_leader():
             self.update_client_data()
 
+        if self.charm.state.peer_cluster_relation and self.charm.unit.is_leader():
+            self.update_peer_cluster_data()
+
```

Comment:
**todo:** (non-blocking) I think there's a bug with something, where `secret-changed` isn't causing `relation-changed`. Not for this PR, this is fine for now, but I think we shouldn't have this line long-term, and should address the `secret-changed` issue itself. I noticed something similar with setting peer-relation secrets not causing a relation-changed also.